### PR TITLE
PostgresSQL Event native support fixes #1719

### LINKED ISSUE #1719: PostgresSQL Event native support

### PR DESCRIPTION
**Pull request by zbendhiba**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // consumer with datasource
+        from("pgevent:///postgres/testchannel?datasource=#pgDataSource")
+                .log("Message got ${body}")
+                .bean(MyBean.class);
+    }
+
+    @Produces
+    @Named("pgDataSource")
+    public PGDataSource loadDataSource() throws Exception {
+        PGDataSource dataSource = new PGDataSource();
```
Comment (jamesnetherton) [inline integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: Maybe it'd be good to leverage the [Quarkus agroal](https://quarkus.io/guides/datasource) extension to create a named postgresql datasource? Or at least add another route / test case for that scenario. It's a good way of proving our stuff integrates well with Quarkus.

**Review by oscerd — APPROVED** [review]

File: extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import org.jboss.logging.Logger;
+
+class PgeventProcessor {
+
+    private static final String PGEVENT_SERVICE_BASE = "META-INF/services/";
+
+    private static final Logger LOG = Logger.getLogger(PgeventProcessor.class);
```
Comment (jamesnetherton) [inline extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java:39]: I think this leftover can be removed.

File: extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    private static final String FEATURE = "camel-pgevent";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerReflectiveClasses() {
```
Comment (jamesnetherton) [inline extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java:48]: If there's just one `ReflectiveClassBuildItem` to return then I guess we don't need a `List`.

File: integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java (hunk 1 above)
Comment (zbendhiba) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: good point!!

Comment (zbendhiba) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: @jamesnetherton well it's not the same datasource. This is datasource expected is the one from pgevent API. The one having a connection that enables us to have those events.

Comment (zbendhiba) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: if we want to have the agroal datasource, we need to convert this one to com.impossibl.postgres.jdbc.PGDatasource

File: extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java (hunk 2 above)
Comment (zbendhiba) [reply extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java]: done

File: extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java (hunk 3 above)
Comment (zbendhiba) [reply extensions/pgevent/deployment/src/main/java/org/apache/camel/quarkus/component/pgevent/deployment/PgeventProcessor.java]: done

File: integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java (hunk 1 above)
Comment (jamesnetherton) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: Ok, maybe we merge the PR as-is and create a follow up issue to look into this afterwards.

I think it should be possible to do. Both `PGDataSource` and `AgroalDatasource` extend `javax.sql.DataSource`. So doing `datasource=#nameOfTheAgroalDatasource` should work (in theory).

Comment (zbendhiba) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: @jamesnetherton yes, I've tested and it works fine. But in our camel component we get the connection from that datasource and if it's not an instance of PGConnection, we raise an error. 
We need to handle a connection of type java.sql.Connection first here : https://github.com/apache/camel/blob/d23efa9cf87f6f5dab886206a3c00cc5e0e2472c/components/camel-pgevent/src/main/java/org/apache/camel/component/pgevent/PgEventHelper.java#L35
So maybe follow up an issue for an improvement.

**Review by jamesnetherton — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]

File: integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java (hunk 1 above)
Comment (zbendhiba) [reply integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java:62]: I've created the issue : https://github.com/apache/camel-quarkus/issues/1909, to follow up

**Pull request merged**